dlib/matrix/matrix_fft_abstract.h

```diff
@@ -21,16 +21,36 @@ namespace dlib
     !*/
 
 // ----------------------------------------------------------------------------------------
+   
+    constexpr long fftr_nc_size(long nc)
+    {
+        return nc == 0 ? 0 : nc/2+1;
+    }
+    /*!
+        ensures
+            - returns the output dimension of a 1D real FFT
+    !*/
```

Comment:
Please move all the `/*! !*/` comments before the `{}` so it looks like the rest of dlib.